lua/telescope/utils.lua

```diff
@@ -489,6 +489,13 @@ utils.get_devicons = load_once(function()
   end
 end)
 
+--- Checks if treesitter parser for language is installed
+---@param lang string
+utils.has_ts_parser = function(lang)
+  local parsers = vim.api.nvim_get_runtime_file("parser/" .. lang .. ".*", false)
```

Comment:
Depends; I use tree-sitter highlighting only very sparsely, so the miss is much more likely than the hit. And this fires on every file I browse through in the picker.

Reply:
>  so the miss is much more likely than the hit

I don't think `language.add` is any slower on misses, but it will make hits a lot quicker. The slowest part is `nvim_get_runtime_file`.

Reply:
Hmm, it validates and registers. I agree on the hit, of course, so it may be worth the tradeoff.